bal_tools/pools_gauges.py

```diff
@@ -236,54 +270,112 @@ def has_alive_preferential_gauge(self, pool_id: str) -> bool:
 
-        # remove pools from blacklist
         blacklist = requests.get(f"{GITHUB_RAW_CONFIG}/core_pools_blacklist.json")
         blacklist.raise_for_status()
         blacklist = blacklist.json()
-        try:
-            for pool in blacklist[self.chain]:
-                if pool in core_pools:
-                    del core_pools[pool]
-        except KeyError:
-            # no results for this chain
-            pass
 
-        return CorePools(pools={PoolId(k): Symbol(v) for k, v in core_pools.items()})
+        chains = core_pools.keys() if return_all_chains else [self.chain]
+        for chain in chains:
+            # sort pools alphabetically by id
+            core_pools[chain] = dict(sorted(core_pools[chain].items()))
+
+            # add pools from whitelist
+            try:
+                for pool, symbol in whitelist[chain].items():
+                    if pool not in core_pools[chain]:
+                        core_pools[chain][pool] = symbol
+            except KeyError:
+                # no results for this chain
+                pass
+
+            # remove pools from blacklist
+            try:
+                for pool in blacklist[chain]:
+                    if pool in core_pools[chain]:
+                        del core_pools[chain][pool]
+            except KeyError:
+                # no results for this chain
+                pass
+
+        if return_all_chains:
+            return core_pools
+        else:
+            return CorePools(
+                pools={PoolId(k): Symbol(v) for k, v in core_pools[self.chain].items()}
+            )
+
+    def filter_core_pool_candidates(self, candidates: List[str]) -> List[str]:
+        """
+        filter a list of core pool candidates based on:
+        - having a tvl of >$100k
+        - being a boosted pool OR
+        - having a non zero rate provider
+        """
+        data = self.subgraph.fetch_graphql_data(
+            "apiv3",
+            "get_core_pools_filters",
+            {"where": {"minTvl": 100_000, "idIn": candidates}},
+        )
+        core_pools = []
+        for pool in data["poolGetPools"]:
+            if "BOOSTED" in pool["tags"]:
+                # v3 boosted pools are always core pools
+                core_pools.append(pool)
+                continue
+
+            for pool_token in pool["poolTokens"]:
+                if pool_token["isExemptFromProtocolYieldFee"] == True:
+                    # pools that are yield fee exempt are not core pools
+                    break
+            else:
+                for pool_token in pool["poolTokens"]:
+                    if pool_token.get("priceRateProviderData"):
+                        if (
+                            pool_token["priceRateProviderData"].get("address")
+                            != ZERO_ADDRESS
+                        ):
+                            # pools with a non zero rate provider are core pools
+                            core_pools.append(pool)
```

Comment:
shouldnt the check be for at least for half of tokens to have rate provider based on [BIP-734](https://forum.balancer.fi/t/bip-734-balancer-v3-launch-and-protocol-enhancements/6168)?
> Minimum 50% yield-bearing or boosted tokens for pool types such as weighted or composable stable pools